core/iwasm/common/wasm_runtime_common.c

```diff
@@ -7071,3 +7072,158 @@ wasm_runtime_detect_native_stack_overflow_size(WASMExecEnv *exec_env,
+
+                if (cur->section_body != NULL)
+                    wasm_runtime_free(cur->section_body);
+                wasm_runtime_free(cur);
+                break;
+            }
+            else {
+                prev = cur;
+                cur = next;
+            }
+        }
+    }
+}
+#endif
+
+#if WASM_ENABLE_AOT != 0
+static void
+destroy_all_aot_sections(aot_section_list_t sections)
+{
+    aot_section_t *cur = sections;
+    while (cur) {
+        aot_section_t *next = cur->next;
+        if (cur->section_body != NULL) {
+            if (cur->section_type == AOT_SECTION_TYPE_TEXT)
+                os_munmap(cur->section_body, cur->section_body_size);
+            else
+                wasm_runtime_free(cur->section_body);
+        }
+        wasm_runtime_free(cur);
+        cur = next;
+    }
+}
+
+static void
+destroy_part_aot_sections(aot_section_list_t *p_sections, uint8 *section_types,
+                          int section_cnt)
+{
+    int i;
+    for (i = 0; i < section_cnt; i++) {
+        uint8 section_type = section_types[i];
+        aot_section_t *cur = *p_sections, *prev = NULL;
+
+        while (cur) {
+            aot_section_t *next = cur->next;
+            if (cur->section_type == section_type) {
+                if (prev)
+                    prev->next = next;
+                else
+                    *p_sections = next;
+
+                if (cur->section_body != NULL) {
+                    if (cur->section_type == AOT_SECTION_TYPE_TEXT)
+                        os_munmap(cur->section_body, cur->section_body_size);
+                    else
+                        wasm_runtime_free(cur->section_body);
+                }
+                wasm_runtime_free(cur);
+                break;
+            }
+            else {
+                prev = cur;
+                cur = next;
+            }
+        }
+    }
+}
+#endif
+
+void
+wasm_runtime_destroy_sections(load_section_result_t *section_result,
+                              uint8_t *section_types, int section_cnt)
+{
+    bool destroy_all_sections = (section_types == NULL);
+    if (destroy_all_sections) {
+#if WASM_ENABLE_INTERP != 0
+        if (!section_result->is_aot)
+            destroy_all_wasm_sections(section_result->sections);
+#endif
+#if WASM_ENABLE_AOT != 0
+        if (section_result->is_aot)
+            destroy_all_aot_sections(section_result->sections);
+#endif
+    }
+    else {
+#if WASM_ENABLE_INTERP != 0
+        if (!section_result->is_aot)
+            destroy_part_wasm_sections(&section_result->sections, section_types,
+                                       section_cnt);
+#endif
+#if WASM_ENABLE_AOT != 0
+        if (section_result->is_aot)
+            destroy_part_aot_sections(&section_result->sections, section_types,
+                                      section_cnt);
+#endif
+    }
+}
+
+bool
+wasm_runtime_read_to_sections(uint8_t *buf, uint64_t size,
+                              load_section_result_t *p_section_result)
```

Comment:
Had better add argument `char *error_buf, uint32 error_buf_size` to this API like other APIs.